src/containers/reCaptcha/reCaptcha.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useDebugValue,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import Script, { type ScriptProps } from "next/script.js";
+
+import { getRecaptchaScriptSrc } from "./utils.js";
+import { type IReCaptcha } from "~/types/recaptcha";
+import { env } from "~/env.mjs";
+
+interface ReCaptchaContextProps {
+  /** reCAPTCHA_site_key */
+  readonly reCaptchaKey: string | null;
+  /** Global ReCaptcha object */
+  readonly grecaptcha: IReCaptcha | null;
+  /** Is ReCaptcha script loaded */
+  readonly loaded: boolean;
+  /** Is ReCaptcha failed to load */
+  readonly error: boolean;
+}
+
+const ReCaptchaContext = createContext<ReCaptchaContextProps>({
+  reCaptchaKey: null,
+  grecaptcha: null,
+  loaded: false,
+  error: false,
+});
+
+const useReCaptchaContext = () => {
+  const values = useContext(ReCaptchaContext);
+  useDebugValue(`grecaptcha available: ${values?.loaded ? "Yes" : "No"}`);
+  useDebugValue(
+    `ReCaptcha Script: ${values?.loaded ? "Loaded" : "Not Loaded"}`,
+  );
+  useDebugValue(`Failed to load Script: ${values?.error ? "Yes" : "No"}`);
+  return values;
+};
+
+interface ReCaptchaProviderProps extends Partial<Omit<ScriptProps, "onLoad">> {
+  reCaptchaKey?: string;
+  language?: string;
+  useRecaptchaNet?: boolean;
+  useEnterprise?: boolean;
+  children?: React.ReactNode;
+  onLoad?: (grecaptcha: IReCaptcha, e: undefined) => void;
+}
+
+const ReCaptchaProvider: React.FC<ReCaptchaProviderProps> = ({
+  reCaptchaKey: passedReCaptchaKey,
+
+  useEnterprise = false,
+  useRecaptchaNet = false,
+  language,
+  children,
+
+  id = "google-recaptcha-v3",
+  strategy = "beforeInteractive",
+
+  src: passedSrc,
+  onLoad: passedOnLoad,
+  onError: passedOnError,
+
+  ...props
+}) => {
+  const [grecaptcha, setGreCaptcha] = useState<IReCaptcha | null>(null);
```

Comment:
```suggestion
  const [grecaptcha, setGreCaptcha] = useState<IReCaptcha>();
```
will default to undefined an the type will automatically include the undefined case